yktrace/src/tir.rs

```diff
@@ -216,6 +286,67 @@ impl TirTrace {
         Ok(Self { ops })
     }
 
+    fn rename_rvalue(rename_map: &HashMap<Local, Vec<Local>>, rvalue: &Rvalue) -> Rvalue {
+        match rvalue {
+            Rvalue::Use(op) => {
+                let newop = TirTrace::rename_operand(rename_map, op);
+                Rvalue::Use(newop)
+            },
+            Rvalue::BinaryOp(binop, op1, op2) => {
+                let newop1 = TirTrace::rename_operand(rename_map, op1);
+                let newop2 = TirTrace::rename_operand(rename_map, op2);
+                Rvalue::BinaryOp(binop.clone(), newop1, newop2)
+            }
+            Rvalue::CheckedBinaryOp(binop, op1, op2) => {
+                let newop1 = TirTrace::rename_operand(rename_map, op1);
+                let newop2 = TirTrace::rename_operand(rename_map, op2);
+                Rvalue::CheckedBinaryOp(binop.clone(), newop1, newop2)
+            }
+            Rvalue::Unimplemented(_) => {
+                rvalue.clone()
+            }
+        }
+    }
+    fn rename_operand(rename_map: &HashMap<Local, Vec<Local>>, operand: &Operand) -> Operand {
+        match operand {
+            Operand::Place(p) => {
+                Operand::Place(TirTrace::rename_place(rename_map, p))
+            }
+            Operand::Constant(_) => {
+                operand.clone()
+            }
+        }
+    }
+
+    fn rename_place(rename_map: &HashMap<Local, Vec<Local>>, place: &Place) -> Place {
+        // In the future there should always be a mapping for any local in the trace.
+        // Unfortunately, since we are still getting remnants of the trace header and tail in our
+        // trace, this is not always the case. So for now print warnings and don't attempt to
+        // rename those locals.
+        match rename_map.get(&place.local) {
+            Some(v) => {
+                match v.last() {
+                    Some(l) => {
+                        let mut p = place.clone();
+                        p.local = l.clone();
+                        p
+                    None => {
+                        println!("warning: mapping for {} already empty", &place.local);
+                        place.clone()
+                    }
+                }
+            }
+            None => {
+                if &place.local != &Local(0) {
+                    // Local(0) is used for function returns and is thus never
+                    // renamed, so we don't need to print a warning for it.
+                    println!("warning: could not find mapping for {}", &place.local);
```

Comment:
I am generally opposed to stdout (or stderr) stuff like this: either this is a condition we should crash over (assert/panic/whatever) to warn us to fix it; or it's not a problem and execution can continue. `print`s are a horrible halfway house :)

Reply:
Maybe use the `log` crate? You can then use `log::warn!("could not find mapping for {}", &place.local);`. To show them you need to call `env_logger::setup();` from the `env_logger` crate **and** set the `RUST_LOG` env var to one of `warn`, `info`, `debug` or `trace`.

Reply:
In this case, it seems to me that if we can't find a mapping, we've somehow got a serious internal consistency error which we need to fix? That doesn't feel like a warning to me, but I'm prepared to be convinced otherwise.

Reply:
The problem is that if we crash our tests won't work anymore and we can't continue working on the compiler. This is something we need to fix in the future (i.e. clean up the trace from the remnants of the start/stop tracing calls), so I thought a warning will make sure we don't forget. I'm also ok with putting a `FIXME` here, or used `log` as @bjorn3 suggested.

Reply:
How many tests fail? Should we mark them as temporarily failing?

Reply:
All of them. Basically our current TIR traces have stuff in them we don't want/need, but getting rid of this is something I will need to look into with @vext01 at some point the future.

Reply:
OK, fair enough. Let's do `eprintln` at least then.

Reply:
Fixed in 4ed196a